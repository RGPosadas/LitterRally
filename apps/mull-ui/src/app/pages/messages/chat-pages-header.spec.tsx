import { MockedProvider } from '@apollo/client/testing';
import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import React from 'react';
import { Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ChatPagesHeader from './chat-pages-header';

describe('ChatPagesHeader', () => {
  it('should render successfully', () => {
    const history = createMemoryHistory();
    const { baseElement } = render(
      <MockedProvider>
        <Router history={history}>
          <ChatPagesHeader>
            <div className="page-container with-sub-nav-and-header with-bottom-chat-input"></div>
          </ChatPagesHeader>
        </Router>
      </MockedProvider>
    );

    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const history = createMemoryHistory();
    const tree = renderer
      .create(
        <MockedProvider>
          <Router history={history}>
            <ChatPagesHeader>
              <div className="page-container with-sub-nav-and-header with-bottom-chat-input"></div>
            </ChatPagesHeader>
          </Router>
        </MockedProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
