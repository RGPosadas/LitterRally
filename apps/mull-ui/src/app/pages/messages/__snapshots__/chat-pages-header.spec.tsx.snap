// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatPagesHeader should match snapshot 1`] = `
<div>
  <div
    className="header undefined"
  >
    <button
      className="hamburger-button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-bars fa-w-14 fa-2x hamburger-menu"
        color="grey"
        data-icon="bars"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <h1>
      Clean up Rogers Park
    </h1>
  </div>
  <div
    className="sub-nav-bar-container with-header top-nav-bar-shadow"
  >
    <div
      className="inner-cont"
    >
      <a
        aria-current={null}
        className="subnavigation-link"
        data-testid="subnavigation-groupchat-button"
        href="/messages/group-chat"
        onClick={[Function]}
      >
        Group Chat
      </a>
      <a
        aria-current={null}
        className="subnavigation-link"
        data-testid="subnavigation-announcements-button"
        href="/messages/announcements"
        onClick={[Function]}
      >
        Announcements
      </a>
    </div>
  </div>
  <div
    className="page-container with-sub-nav-and-header with-bottom-chat-input"
  />
</div>
`;
